include/pthread.h

```diff
@@ -0,0 +1,376 @@
+#ifndef _PTHREAD_H
+#define _PTHREAD_H
+
+/* FIXME: This should probably live somewhere else but Zephyr doesn't
+ * currently have a stdc layer to provide it
+ */
+struct timespec {
+	s32_t tv_sec;
+	s32_t tv_nsec;
+};
+
+static inline s32_t _ts_to_ms(const struct timespec *to)
+{
+	return (to->tv_sec * 1000) + (to->tv_nsec / 1000000);
+}
+
+typedef struct pthread_mutex {
+	struct k_sem *sem;
```

Comment:
Oh, and to repeat the point in a separate comment: honestly we can probably skip the syscall treatment for these.  These aren't native APIs, they're going to be used by people porting code into Zephyr from existing code bases.  Code using pthreads in the wild is going to be threaded: by design it won't rely on memory protection between the threads (and in fact default pthreads doesn't even support that use case: you need an implementation that supports the "process shared" attribute that got added to the standard fairly recently).